aggregate.go

```diff
@@ -5,11 +5,32 @@ import (
 	"sort"
 
 	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
 )
 
+type AggregatorConfig struct {
+	Logger micrologger.Logger
+}
+
+type Aggregator struct {
+	logger micrologger.Logger
+}
+
+func NewAggregator(config AggregatorConfig) (*Aggregator, error) {
```

Comment:
This makes the aggregator an object which we can configure with logging later on. More to come in upcoming PRs.